components/datadog/apps/tracegen/images/tracegen/main.go

```diff
@@ -59,6 +73,16 @@ func main() {
 		}
 	}
 
+	var opts []tracer.StartOption
+	if v, ok := os.LookupEnv("ECS_AGENT_HOST"); ok && v == "true" {
+		host, err := retrieveDDAgentHostECS()
+		if err != nil {
+			panic(fmt.Sprintf("Failed to retrieve DD agent host: %v", err))
+		}
```

Comment:
Is this reliable? Can we retry to avoid flakiness?

Reply:
Also can you please add a comment to explain and perhaps add this link https://docs.datadoghq.com/containers/amazon_ecs/apm/?tab=ec2metadataendpoint#code ? thank you!

Reply:
As discussed offline, using the metadata file might be a better / more robust approach https://docs.datadoghq.com/containers/amazon_ecs/apm/?tab=ecscontainermetadatafile#code